core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -52,6 +56,10 @@ oppia.factory('oppiaPlayerService', [
 
     var exploration = null;
 
+    var totalTimeStopwatch = null;
+
+    var visitedStates = [];
```

Comment:
visitedStateNames. Important to distinguish between the state object and its name.

Reply:
Done.